### PR TITLE
Ensure Spring RestControllers paths show up in dev-mode 404

### LINKED ISSUE #4637: Spring Controller's don't show up in quarkus:dev 404 page
The live reload mode (quarkus:dev) provides a great error page that displays all the endpoints.
However it currently only displays JAX-RS endpoints and should be extended to support Spring Controllers as well

### PR DESCRIPTION
**Pull request by geoand**
Fixes #4637

### CODE REVIEW

**Comment by geoand** [comment]
@gsmet am I too late for `0.26` :P?

File: extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NotFoundExceptionMapper.java [hunk 1]
```diff
@@ -45,6 +45,7 @@
 
     private volatile static List<String> servletMappings = Collections.EMPTY_LIST;
     private volatile static List<String> staticResources = Collections.EMPTY_LIST;
+    private volatile static Map<String, NonJaxRsClassMappings> nonJaxRsPaths = Collections.EMPTY_MAP;
```
Comment (FroMage) [inline extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NotFoundExceptionMapper.java]: I would rather have a name that indicates that we're mapping from class names, which appears to be the case.

**Review by FroMage — CHANGES_REQUESTED** [review]
LGTM except for one ambiguous variable name,

File: extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NotFoundExceptionMapper.java (hunk 1 above)
Comment (geoand) [reply extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NotFoundExceptionMapper.java]: Makes sense, will fix.

File: extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/ExceptionMapperRecorder.java [hunk 2]
```diff
@@ -17,4 +17,15 @@ public void setServlets(Map<String, List<String>> servletToMapping) {
...
     }
 
+    /**
+     * Uses to register the paths of classes that are not annotated with JAX-RS annotations (like Spring Controllers for
+     * example)
+     * 
+     * @param nonJaxRsPaths A that contains the class name as a key and class that contains the base path and a map that
+     *        contains the method name to path as a value
+     */
+    public void setNonJaxRsPaths(Map<String, NonJaxRsClassMappings> nonJaxRsPaths) {
```
Comment (FroMage) [inline extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/ExceptionMapperRecorder.java]: Actually, same here: I'd use a variable name that makes the javadoc non-required.

Comment (geoand) [reply extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/ExceptionMapperRecorder.java]: :+1:

File: extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NotFoundExceptionMapper.java (hunk 1 above)
Comment (geoand) [reply extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NotFoundExceptionMapper.java]: Better now?

File: extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/ExceptionMapperRecorder.java (hunk 2 above)
Comment (geoand) [reply extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/ExceptionMapperRecorder.java]: Any better?

**Review by FroMage — APPROVED** [review]
Thanks.

**Pull request merged**